public/utils/constants.ts

```diff
@@ -59,21 +59,6 @@ export const MAX_DETECTORS = 1000;
 
 export const MAX_ANOMALIES = 10000;
 
-export enum DETECTOR_STATE {
-  DISABLED = 'Stopped',
-  INIT = 'Initializing',
-  RUNNING = 'Running',
-  FEATURE_REQUIRED = 'Feature required',
-  INIT_FAILURE = 'Initialization failure',
-  UNEXPECTED_FAILURE = 'Unexpected failure',
-}
-
-export enum SAMPLE_TYPE {
-  HTTP_RESPONSES = 'http-responses',
-  HOST_HEALTH = 'host-health',
-  ECOMMERCE = 'ecommerce',
-}
```

Comment:
Any reason for just moving these constants to the server side? Is it because these are the only ones shared between client and server side?

Reply:
when building the artifact, the source code of `public` folder is ignored: https://github.com/elastic/kibana/blob/v7.10.0/packages/kbn-plugin-helpers/src/tasks/write_server_files.ts#L66

Assuming we move these constants to client side, and customer builds the artifact and installs it, kibana will fail to start because such constants are missing in the artifact.

Reply:
I see, makes sense. Thanks for the explanation